### PR TITLE
feat: add appPath to handleCreateSession

### PR DESCRIPTION
**Pull request by proAlexandr**
TL;DR: Allows to explicitly specify path to .app application that will be opened by driver.

Context:
During development you can have many build of the same application with the same bundleId on your machine. You have no control which one will be opened by `[[XCUIApplication alloc] initWithBundleIdentifier:bundleID]`. So you can end up testing wrong build (for example, production build from /Application folder instead of dev build that includes local changes).

This PR introduces appPath argument for /session request to specify a path to .app application and be sure that webdriver launches exactly the same build that you need.

### CODE REVIEW

File: WebDriverAgentMac/WebDriverAgentLib/Commands/FBSessionCommands.m [hunk 1]
```diff
@@ -87,7 +88,13 @@ + (NSArray *)routes
   if (nil == bundleID) {
     session = [FBSession initWithApplication:nil];
   } else {
-    XCUIApplication *app = [[XCUIApplication alloc] initWithBundleIdentifier:bundleID];
+    NSString *appPath = requirements[AM_APP_PATH_CAPABILITY];
+    XCUIApplication *app;
```
Comment (mykola-mokhnach) [inline WebDriverAgentMac/WebDriverAgentLib/Commands/FBSessionCommands.m:92]: use a ternary operator instead

File: WebDriverAgentMac/WebDriverAgentLib/Utilities/AMSessionCapabilities.m [hunk 2]
```diff
@@ -21,3 +21,4 @@
 NSString* const AM_APP_ENVIRONMENT_CAPABILITY = @"environment";
 NSString* const AM_SKIP_APP_KILL_CAPABILITY = @"skipAppKill";
 NSString* const AM_NO_RESET_CAPABILITY = @"noReset";
+NSString* const AM_APP_PATH_CAPABILITY = @"appPath";
```
Comment (mykola-mokhnach) [inline WebDriverAgentMac/WebDriverAgentLib/Utilities/AMSessionCapabilities.m:24]: the new capability should be defined in https://github.com/appium/appium-mac2-driver/blob/master/lib/desired-caps.js and documented in README

File: WebDriverAgentMac/WebDriverAgentLib/Commands/FBSessionCommands.m [hunk 3]
```diff
@@ -87,7 +88,13 @@ + (NSArray *)routes
...
+    if (appPath == nil) {
+      app = [[XCUIApplication alloc] initWithBundleIdentifier:bundleID];
+    } else {
+      app = [[XCUIApplication alloc] initPrivateWithPath:appPath bundleID:bundleID];
```
Comment (mykola-mokhnach) [inline WebDriverAgentMac/WebDriverAgentLib/Commands/FBSessionCommands.m:96]: should we validate if appPath is actually a valid path to an app bundle and throw an exception if it is not?

**Comment by mykola-mokhnach** [comment]
@proAlexandr Please sign the CLA

File: WebDriverAgentMac/WebDriverAgentLib/Commands/FBSessionCommands.m (hunk 3 above)
Comment (proAlexandr) [reply WebDriverAgentMac/WebDriverAgentLib/Commands/FBSessionCommands.m:96]: I think we can skip validation here since it happens inside [XCUIApplication  initPrivateWithPath:bundleID].
If a path is not valid it will throw an error with message: "Failed to determine bundle identifier for '/Users/alex/program/out/SomeApp.app'. The application may be damaged or incomplete."

**Comment by proAlexandr** [comment]
Hi @mykola-mokhnach. Thanks a lot for review!
I have signed SLA and updated the PR according to comments.

File: README.md [hunk 4]
```diff
@@ -47,6 +47,7 @@ appium:skipAppKill | Whether to skip the termination of the application under te
...
 appium:postrun | An object containing either `script` or `command` key. The value of each key must be a valid AppleScript script or command to be executed after Mac2Driver session is stopped. See [AppleScript commands execution](#applescript-commands-execution) for more details.
 appium:noReset | Whether to restart the app whose bundle identifier was passed to capabilities as `bundleId` value if it was already running on the session startup (`false`, the default value) or just pick it up without changing the app state (`true`). Note that neither of `arguments` or `environment` capabilities will take effect if the app did not restart.
+appium:appPath | The path to the application to automate, for example `/Applications/MyAppName.app`. This is an optional capability, but it requires `bundleId` to be present. If `bundleId` is empty, `appPath` will be ignored. If the path is invalid or application is damaged/incomplete then an error will be thrown on session startup.
```
Comment (mykola-mokhnach) [inline README.md:50]: we probably need an explanation here on why this capability might be useful together with bundleId

**Review by mykola-mokhnach — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (proAlexandr) [reply README.md]: Updated README

File: README.md [hunk 5]
```diff
@@ -47,7 +47,7 @@ appium:skipAppKill | Whether to skip the termination of the application under te
...
-appium:appPath | The path to the application to automate, for example `/Applications/MyAppName.app`. This is an optional capability, but it requires `bundleId` to be present. If `bundleId` is empty, `appPath` will be ignored. If the path is invalid or application is damaged/incomplete then an error will be thrown on session startup.
+appium:appPath | The path to the application to automate, for example `/Applications/MyAppName.app`. This is an optional capability, but it requires `bundleId` to be present. If `bundleId` is empty, `appPath` will be ignored. If the path is invalid or application is damaged/incomplete then an error will be thrown on session startup. This capability could be usefull when you have multiple builds of the same application with the same bundleId on your machine (for example one production build in /Applications/ and several dev build). If you provide bundleId only, operation system could open any of these builds. By providing `appPath` you have a guaranty that specified .app will be launched, not arbitrary one.
```
Comment (mykola-mokhnach) [inline README.md:50]: operation system -> the operating system
you have a guaranty -> you have a guarantee
that specified .app will -> that the specified .app will
not arbitrary one -> rather than a random one
and several dev build -> and several dev builds
